store/tikv/region_request.go

```diff
@@ -178,6 +191,32 @@ func (s *RegionRequestSender) sendReqToRegion(bo *Backoffer, ctx *RPCContext, re
 	return
 }
 
+func (s *RegionRequestSender) getStoreToken(store *Store) error {
+	for {
+		limit := atomic.LoadUint32(&store.storeLimit)
+		if limit < atomic.LoadUint32(&config.GetGlobalConfig().StoreLimit) {
+			if atomic.CompareAndSwapUint32(&store.storeLimit, limit, limit+1) {
+				return nil
+			}
+		} else {
+			return ErrTokenLimit.GenWithStackByArgs("can't get store token, pool is up to the limit")
+		}
+	}
+}
+
+func (s *RegionRequestSender) releaseStoreToken(store *Store) error {
+	for {
+		limit := atomic.LoadUint32(&store.storeLimit)
+		if limit > 0 {
+			if atomic.CompareAndSwapUint32(&store.storeLimit, limit, limit-1) {
+				return nil
+			}
+		} else {
+			return ErrTokenLimit.GenWithStackByArgs("can't release store token, pool is full")
```

Comment:
pool is full? Maybe you should rephrase this message, it's confusing.
BTW, In which situation will this happen? When there's a BUG?

Reply:
Addressed.